torch/_functorch/functional_call.py

```diff
@@ -103,3 +103,52 @@ def compute_loss(params, x, t):
         parameters_and_buffers = {k: v for d in parameter_and_buffer_dicts for k, v in d.items()}
 
     return nn.utils.stateless.functional_call(module, parameters_and_buffers, args, kwargs)
+
+
+@exposed_in("torch.func")
+def stack_module_state(models):
+    """stack_module_state(models) -> params, buffers
+
+    Prepares a list of torch.nn.Modules for ensembling with :func:`vmap`.
+
+    Given a list of ``M`` ``nn.Modules`` of the same class, returns two dictionaries
+    that stack all of their parameters and buffers together, indexed by name.
```

Comment:
nit: maybe add an example with submodules that prints out the dictionaries so the user can see the convention of the name strings?

Reply:
added!